perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -333,14 +341,23 @@ int main(int ArgCount, char** Args)
     instruction_table Table;
     Sim86_Get8086InstructionTable(&Table);
     printf("8086 Instruction Instruction Encoding Count: %u\n", Table.EncodingCount);
-
+    
+    bool DumpFile = false;
+    
     for (int ArgIndex = 1; ArgIndex < ArgCount; ArgIndex++)
     {
+        std::string FileName = Args[ArgIndex];
+        if (FileName == "-dump")
+        {
+            DumpFile = true;
+            continue;
+        }
```

Comment:
This is not a very robust implementation of options because if you pass one file, then `-dump`, then one file, it will only work for the second one. This is allowed by your implementation.

Command line programs usually work a slightly different way: all options have to be at the beginning, or at the end, but are not mixed with arguments which are not options. There are many choices on how to do this, in addition to what I already mentioned, you could pass the files using an option like Muratori does with `-exec`, or you can choose to mix everything as well and do it your way, I think it's fine. :D Just saying this to let you know of the usual conventions.

Reply:
Thanks! I assumed all arguments had to go at the beginning but somehow failed to consider it was up to me to enforce it. 76e807c1f60c9898ff0c7543f65c4056b0e17b01